terraso_backend/apps/auth/views.py

```diff
@@ -133,15 +135,18 @@ def post(self, request, *args, **kwargs):
 
 class CheckUserView(View):
     def get(self, request, *args, **kwargs):
-        if not request.user.is_authenticated:
+        user = request.user
+        profile_image_service = ProfileImageService()
+        if not user.is_authenticated:
             return JsonResponse({"error": "Unauthenticated request."}, status=401)
 
         return JsonResponse(
             {
                 "user": {
-                    "email": request.user.email,
-                    "first_name": request.user.first_name,
-                    "last_name": request.user.last_name,
+                    "email": user.email,
+                    "first_name": user.first_name,
+                    "last_name": user.last_name,
+                    "profile_image": profile_image_service.get_image_url(user.profile_image_path),
```

Comment:
The `user.profile_image_path` hasn't already the built image URL? Why to call the service method here?

Reply:
@cacarrara To avoid being tightly coupled to S3 or the the domain from where we can access the images, the only thing we will be stored in the DB is the path to the image inside the source of truth of the files. That is why we need to create the complete URL here

Reply:
I was thinking to persist the entire URL exactly to be decoupled from any specific storage. Once we uploaded the file and got the URL at the upload moment, we don't need to care about the source of the truth of the files anymore. Let's suppose we start working with S3 now and at some point we decide to start uploading image to `imgur` (absurd example). Then, with this implementation, we got stuck to S3, or we have to work around on application level to build the proper URL for the other service.

Despite the point above, let's suppose we have to dump the DB or export the DB data to any format, like CSV for any reason. Then, we wouldn't have the complete profile image URL as well, unless we pass through the application layer.

What do you think about that?

Reply:
I agree that we should store the full URL, for the reasons Caio states.

Reply:
I see, I didn't think about those potential scenarios, so lets gather all of our concerns:
1. We might want to change the current storage for user images
2. We might want to use more than one storage location for user images
3. The DB doesn't have the full location of the user images, so we have to always relay on the application layer to identify the complete URL.

Lets do a simplistic "risk" analysis exercise of the potential scenarios if those arise in the future with the 2 approaches:

Effort: (0-5(high)) Effort to change the implementation to support this
Likelihood: (0-5(high)) Likelihood that the future scenario will appear

**Current Implementation: Persist only image path** (value: 38)

1. We might want to change the current storage for user images (value: 6)
  Effort: 1 (Change the constants, move files to new location)
  Likelihood: 6 (Having in mind that we don't have our infrastructure in AWS right now it is likely that we don't want to use S3 in the future)
2. We might want to use more than one storage location for user images (value: 8)
  Effort: 4 (Create script to complete URL appending the new base path, add support for extra storage)
  Likelihood: 2 (Seems that this scenario will be really unlikely, I don't se a reason to support multiple storages at the top of my mind)
3. DB is not the source of truth for files URLs, and we need to gather data from there directly (value:  24)
  Effort: 3 (Besides the script to generate the data we need an extra step to complete the URLs)
  Likelihood: 8 (It seems really possible that we might want to extract data directly from the DB)
  
**Potential implementation: Persist complete URL** (value: 28)

1. We might want to change the current storage for user images (value: 24)
  Effort: 4 (Create a script to gather all persisted URLs change them and save them again with the new URL, move files to the new location)
  Likelihood: 6 (Having in mind that we don't have our infrastructure in AWS right now it is likely that we don't want to use S3 in the future)
2. We might want to use more than one storage location for user images (value: 4)
  Effort: 2 (just add support for extra storage)
  Likelihood: 2 (Seems that this scenario will be really unlikely, I don't se a reason to support multiple storages at the top of my mind)
3. DB is not the source of truth for files URLs, and we need to gather data from there directly (value: 0)
  Effort: 0 (Already provided by the solution)
  Likelihood: 8 (It seems really possible that we might want to extract data directly from the DB)
  
@cacarrara It seems that your approach to hold the full URL has less potential risk, but let me know if you see a whole in my logic haha

Reply:
Wow. You went really deep into it :astonished: -- Congrats and thanks for this analysis. My thought always was more about respecting some principles (don't repeat yourself, separation of concerns, single responsibility). Also, keeping the application flexible enough for changes. We cannot predict all possible changes we can have (thanks for the exercise of likelihood). While probably we won't have more than one storage for images, things like migrate the current schema to support CDNs, or even some sort of migration where we can incrementally migrate the DB, seems reasonable. I think having the URL on DB enables this kind of flexibility, respecting the principles.